kale/embed/mpca.py

```diff
@@ -144,52 +144,43 @@ def _fit(self, x):
         proj_mats = []
 
         for i in range(1, n_dims):
-            for j in range(n_samples):
-                # unfold the j_th tensor along the i_th mode
-                x_ji = unfold(x[j, :], mode=i - 1)
-                if i not in phi:
-                    phi[i] = 0
-                phi[i] = phi[i] + np.dot(x_ji, x_ji.T)
+            phi[i] = unfold(x, mode=i)
 
         # get the output tensor shape based on the cumulative distribution of eigenvalues for each mode
         for i in range(1, n_dims):
-            # eig_values, eig_vectors = np.linalg.eig(phi[i])
-            eig_vectors, eig_values, vt = np.linalg.svd(phi[i])
-            idx_sorted = (-1 * eig_values).argsort()
-            cum = eig_values[idx_sorted]
+            u, s, v = linalg.svd(phi[i], full_matrices=False)
+            idx_sorted = (-1 * s).argsort()
+            cum = s[idx_sorted]
             var_tot = np.sum(cum)
 
             for j in range(1, cum.shape[0] + 1):
                 if np.sum(cum[:j]) / var_tot > self.var_ratio:
                     shape_out += (j,)
                     break
-            proj_mats.append(eig_vectors[:, idx_sorted][:, : shape_out[i - 1]].T)
+            proj_mats.append(u[:, idx_sorted][:, : shape_out[i - 1]].T)
```

Comment:
u is not as readable

Reply:
It seems `u`, `s` and `v` are common variable names for SVD in popular python packages like scipy, numpy, scikit-learn, pytorch, and tensorly. That is why `u` is used here. To improve readability, `eig_vectors` or `singular_vectors_left` probably are good choices.

Reply:
I think so, not all know the relationship between svd and eigendecomposition.